flower/command.py

```diff
@@ -48,7 +50,12 @@ def sigterm_handler(signal, frame):
         sys.exit(0)
 
     signal.signal(signal.SIGTERM, sigterm_handler)
+
+    if not is_broker_connected(celery_app=app):
```

Comment:
here in theory the connection is checked but can drop before we call flower.start().

The chances are super small though.
I will think about better way of doing this but for now it is good enough.

Reply:
I looked at kombu code:

```python
    def ensure_connection(self, *args, **kwargs):
        """Public interface of _ensure_connection for retro-compatibility.

        Returns kombu.Connection instance.
        """
        self._ensure_connection(*args, **kwargs)
        return self
```

and they follow similar pattern (check connection and then return value) so I think we should be fine with making a check for connection and then calling `flower.start()`